diff --git a/src/error.c b/src/error.c
index 3e5e5e9..997063a 100644
--- a/src/error.c
+++ b/src/error.c
@@ -597,8 +597,8 @@ _IceDefaultIOErrorHandler (
 )
 {
     fprintf (stderr,
-	"ICE default IO error handler doing an exit(), pid = %ld, errno = %d\n",
-	(long)getpid(), errno);
+	"ICE default IO error handler doing an exit(), errno = %d\n",
+	errno);
 
     exit (1);
 }
diff --git a/src/iceauth.c b/src/iceauth.c
index ef66626..d7c05e5 100644
--- a/src/iceauth.c
+++ b/src/iceauth.c
@@ -76,9 +76,8 @@ IceGenerateMagicCookie (
     }
 #else
     {
-	long    time ();
-	ldata[0] = time ((long *) 0);
-	ldata[1] = getpid ();
+	ldata[0] = time (NULL);
+	ldata[1] = time (NULL);
     }
 #endif
     seed = (ldata[0]) + (ldata[1] << 16);
diff --git a/src/process.c b/src/process.c
index f0c3369..28fa34e 100644
--- a/src/process.c
+++ b/src/process.c
@@ -715,7 +715,7 @@ ProcessError (
 	{
 	    _IceProtocolError *errorReply =
 	        &(((_IceReply *) (replyWait->reply))->protocol_error);
-	    char *errorStr = "";
+	    char *errorStr;
 	    const char *prefix;
 	    char *temp;
 
