From 506ddb816962d5951bf5031b93fde6a1e43983d9 Mon Sep 17 00:00:00 2001
From: Jeffrey Clark <dude@zaplabs.com>
Date: Wed, 27 Apr 2016 13:51:13 +0000
Subject: [PATCH 1/2] support for ATSC closed captions

---
 libmpeg2/decode.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/libmpeg2/decode.c b/libmpeg2/decode.c
index 0908782..389cdec 100644
--- a/libmpeg2/decode.c
+++ b/libmpeg2/decode.c
@@ -212,7 +212,7 @@ mpeg2_state_t mpeg2_parse (mpeg2dec_t * mpeg2dec)
 
 mpeg2_state_t mpeg2_parse_header (mpeg2dec_t * mpeg2dec)
 {
-    static int (* process_header[]) (mpeg2dec_t * mpeg2dec) = {
+    static int (* process_header[]) (mpeg2dec_t *) = {
 	mpeg2_header_picture, mpeg2_header_extension, mpeg2_header_user_data,
 	mpeg2_header_sequence, NULL, NULL, NULL, NULL, mpeg2_header_gop
     };
@@ -368,6 +368,14 @@ void mpeg2_slice_region (mpeg2dec_t * mpeg2dec, int start, int end)
 
 void mpeg2_tag_picture (mpeg2dec_t * mpeg2dec, uint32_t tag, uint32_t tag2)
 {
+    if (mpeg2dec->num_tags == 0 && mpeg2dec->state == STATE_PICTURE && mpeg2dec->picture) {
+        // since tags got processed when we entered this state we
+        // have to set them directly or they'll end up on the next frame.
+        mpeg2dec->picture->tag = tag;
+        mpeg2dec->picture->tag2 = tag2;
+        mpeg2dec->picture->flags |= PIC_FLAG_TAGS;
+        return;
+    }
     mpeg2dec->tag_previous = mpeg2dec->tag_current;
     mpeg2dec->tag2_previous = mpeg2dec->tag2_current;
     mpeg2dec->tag_current = tag;
-- 
2.8.0

