Signed-off-by: Jon TURNEY <jon.turney at dronecode.org.uk>
---
 src/Shell.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/Shell.c b/src/Shell.c
index 2a627cc..fe5468a 100644
--- a/src/Shell.c
+++ b/src/Shell.c
@@ -92,6 +92,10 @@ in this Software without prior written authorization from The Open Group.
 #include <X11/Xmu/Editres.h>
 #endif
 
+#ifdef _WIN32
+#include <process.h>            /* for getpid() */
+#endif
+
 /***************************************************************************
  *
  * Note: per the Xt spec, the Shell geometry management assumes in
-- 
2.3.2
